packages/hydrogen/src/cache/api.ts

```diff
@@ -135,7 +131,7 @@ async function setItem(
   // cache-control is still necessary for mini-oxygen
   response.headers.set('cache-control', paddedCacheControlString);
   response.headers.set('real-cache-control', cacheControlString);
-  response.headers.set('cache-put-date', new Date().toUTCString());
+  response.headers.set('cache-put-date', String(Date.now()));
```

Comment:
@wizardlyhel Do you remember why this was using `toUTCString`? It introduces a rounding error (~500ms as mentioned in the note above. Instead, `.toISOString()` could fix the error but we can also just use the timestamp directly I think?

Reply:
I understand the reasoning about the different timezone with `toUTCString` but that function does not return milliseconds so it's impossible to parse it into something accurate (half second rounding error). On the other hand, `toISOString` contains milliseconds:
- `toUTCString`: `Mon, 10 Jun 2024 02:03:08 GMT`
- `toISOString`: `2024-06-10T02:03:08.920Z`

They are both always based on UTC.

In any case, I think a simple [timestamp based on UTC](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/now) is also fine and even simpler, right?

Reply:
But `Date.now()` also considers UTC and is simpler conceptually to store milliseconds than a date. Let's use that better, no?

Reply:
Agreed, I think `Date.now()` is simpler